core/src/subgraph/instance.rs

```diff
@@ -230,14 +232,14 @@ where
         data_source: DataSource,
         top_level_templates: Arc<Vec<DataSourceTemplate>>,
         metrics: Arc<HostMetrics>,
-    ) -> Result<Arc<T::Host>, Error> {
+    ) -> Result<Arc<T::Host>, anyhow::Error> {
         // Protect against creating more than the allowed maximum number of data sources
         if let Some(max_data_sources) = *MAX_DATA_SOURCES {
             if self.hosts.len() >= max_data_sources {
-                return Err(format_err!(
+                anyhow::bail!(
```

Comment:
Since the failure stuff is in the prelude right now, I'd rather not mix it with anyhow stuff since they often have the same name. Once we have more anyhow than failure in the code, we can switch.